### ISSUE TITLE
problem in spyder

### ISSUE THREAD

**Issue by mohamedbahru** [spyder-ide/spyder#17794]
### What steps will reproduce the problem?

I achanged the python to 2.7 and this error starts appearing

### Traceback
```python-traceback
Exception in comms call get_namespace_view:
  File "C:\Users\mohamedb\Anaconda3\envs\py27\lib\site-packages\spyder_kernels\comms\commbase.py", line 346, in _handle_remote_call
    buffer['call_kwargs'])
  File "C:\Users\mohamedb\Anaconda3\envs\py27\lib\site-packages\spyder_kernels\comms\frontendcomm.py", line 264, in _remote_callback
    thread_id = threading.get_ident()
AttributeError: 'module' object has no attribute 'get_ident'
Exception in comms call get_var_properties:
  File "C:\Users\mohamedb\Anaconda3\envs\py27\lib\site-packages\spyder_kernels\comms\commbase.py", line 346, in _handle_remote_call
    buffer['call_kwargs'])
  File "C:\Users\mohamedb\Anaconda3\envs\py27\lib\site-packages\spyder_kernels\comms\frontendcomm.py", line 264, in _remote_callback
    thread_id = threading.get_ident()
AttributeError: 'module' object has no attribute 'get_ident'
Exception in comms call get_cwd:
  File "C:\Users\mohamedb\Anaconda3\envs\py27\lib\site-packages\spyder_kernels\comms\commbase.py", line 346, in _handle_remote_call
    buffer['call_kwargs'])
  File "C:\Users\mohamedb\Anaconda3\envs\py27\lib\site-packages\spyder_kernels\comms\frontendcomm.py", line 264, in _remote_callback
    thread_id = threading.get_ident()
AttributeError: 'module' object has no attribute 'get_ident'
```

## Versions

* Spyder version: 5.3.0 
* Python version: 3.10.4
* Qt version: 5.12.9
* PyQt5 version: 5.12.3
* Operating System: Windows 10

### Dependencies

```
# Mandatory:
atomicwrites >=1.2.0          :  1.4.0 (OK)
chardet >=2.0.0               :  4.0.0 (OK)
cloudpickle >=0.5.0           :  2.0.0 (OK)
cookiecutter >=1.6.0          :  1.7.3 (OK)
diff_match_patch >=20181111   :  20200713 (OK)
intervaltree >=3.0.2          :  3.0.2 (OK)
IPython >=7.31.1;<8.0.0       :  7.32.0 (OK)
jedi >=0.17.2;<0.19.0         :  0.18.1 (OK)
jellyfish >=0.7               :  0.9.0 (OK)
jsonschema >=3.2.0            :  4.4.0 (OK)
keyring >=17.0.0              :  23.4.0 (OK)
nbconvert >=4.0               :  6.5.0 (OK)
numpydoc >=0.6.0              :  1.2.1 (OK)
paramiko >=2.4.0              :  2.10.4 (OK)
parso >=0.7.0;<0.9.0          :  0.8.3 (OK)
pexpect >=4.4.0               :  4.8.0 (OK)
pickleshare >=0.4             :  0.7.5 (OK)
psutil >=5.3                  :  5.9.0 (OK)
pygments >=2.0                :  2.12.0 (OK)
pylint >=2.5.0                :  2.13.7 (OK)
pyls_spyder >=0.4.0           :  0.4.0 (OK)
pylsp >=1.4.1;<1.5.0          :  1.4.1 (OK)
pylsp_black >=1.2.0           :  1.2.1 (OK)
qdarkstyle >=3.0.2;<3.1.0     :  3.0.3 (OK)
qstylizer >=0.1.10            :  0.2.1 (OK)
qtawesome >=1.0.2             :  1.1.1 (OK)
qtconsole >=5.3.0;<5.4.0      :  5.3.0 (OK)
qtpy >=2.0.1                  :  2.0.1 (OK)
rtree >=0.9.7                 :  1.0.0 (OK)
setuptools >=49.6.0           :  62.1.0 (OK)
sphinx >=0.6.6                :  4.5.0 (OK)
spyder_kernels >=2.3.0;<2.4.0 :  2.3.0 (OK)
textdistance >=4.2.0          :  4.2.2 (OK)
three_merge >=0.1.1           :  0.1.1 (OK)
watchdog >=0.10.3             :  2.1.7 (OK)
zmq >=17                      :  22.3.0 (OK)

# Optional:
cython >=0.21                 :  None (NOK)
matplotlib >=3.0.0            :  3.5.1 (OK)
numpy >=1.7                   :  1.22.3 (OK)
pandas >=1.1.1                :  1.4.2 (OK)
scipy >=0.17.0                :  1.8.0 (OK)
sympy >=0.7.3                 :  1.10.1 (OK)
```

**Comment by ccordoba12** [comment]
Hey @mohamedbahru, thanks for reporting. This is a problem in Spyder-kernels that broke compatibility with Python 2. We'll try to fix it in a future version.

**Comment by ccordoba12** [comment]
Hey @mohamedbahru, sorry for the late reply. We plan to fix this in our next version (5.5.0), to be released next month.

For now I'm going to close this issue in favor of #21310 because it has a bit more of context.
